### PR TITLE
Better a11y focus / highlight on header

### PR DESCRIPTION
**Pull request by jimleroyer**
# Summary | Résumé

Better focus highlight, shape and style around various components in the web app, as described [in this Trello card](https://trello.com/c/AtQwiIVe/456-developmentfocus-states-consistency-contrast-and-colours-need-work).

# Test instructions | Instructions pour tester la modification

Load up the web application and tab through all the various elements. If something looks wrong, fill in an task [in the Trello card](https://trello.com/c/AtQwiIVe/456-developmentfocus-states-consistency-contrast-and-colours-need-work).

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/1e61df52239ecdc54a43a7a8ffa639fef2dc66b2

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/935f669ef48dea9048b86169d795865c22162b03

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/7f5a92b333ab05e6d8cada47f8e195631d0eb372

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/fc5cbc9484e57f19a4b8d7d5de2a3ee2b7da32b8

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/4c131ce938eacfa3ce08062cf52633f468c83ab0

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/b9bce64f159b54649d0b7f0cda5967caf3adc48c

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/a3d00c2fb662b120c3696d433d7aba0e48dba0ac

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/acea52d0ec5d27f8ba30213bf7c6cec497dd59e9

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/14d4716a15caecb3cfb28f9608af0410c6004315

File: app/templates/components/nav_menu_item.html [hunk 1]
```diff
@@ -18,7 +18,7 @@
             aria-label="{{ localised_txt }}"
             src="{{ external_link_img }}"
             alt="{{ _('External link') }}"
-            class="inline w-8 h-8 pb-2"/>
+            class="inline w-8 h-8 ml-2 self-center"/>
```
Comment (AntoineAugusti) [inline app/templates/components/nav_menu_item.html:21]: This images "jump" when tabbing in the menu on desktop, can it be fixed?

File: app/templates/partials/nav/gc_header.html [hunk 2]
```diff
@@ -11,20 +13,18 @@
...
       {% if session["userlang"] == "en" %}
-        <a href="/">
-          <img
-            alt="{{ _('Government of Canada') }}"
-            width="250"
-            class=""
-            src="{{ asset_s3_url('gov-canada-en.svg') }}"
-          />
-        </a>
+        <img
```
Comment (AntoineAugusti) [inline app/templates/partials/nav/gc_header.html:16]: Could you refactor the GoC logo code? Should be able to

```py
{{ asset_s3_url('gov-canada-' + session["userlang"] + '.svg') }}
```

Comment (AntoineAugusti) [reply app/templates/partials/nav/gc_header.html:16]: The GoC logo does not have a focus state when it's focused, is it expected?

File: app/templates/components/menu-button.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{% macro menu_button(
+  href,
+  label,
+  aria_label=None,
+  lang=None,
+  style=None
+) %}
+    <a
+        {% if aria_label %} aria-label="{{ aria_label }}" {% endif %}
+        class="inline-block self-center py-1 px-4 line-under text-blue text-small focus:outline-none focus:bg-yellow focus:border-b-2 focus:border-blue link:text-blue visited:text-blue {% if style %} {{ style }} {% endif %}"
```
Comment (AntoineAugusti) [inline app/templates/components/menu-button.html:10]: When tabbing Sign in / language switch, links jump. WDYT about having a transparent bottom border by default and changing the color on focus?

**Review by AntoineAugusti — CHANGES_REQUESTED** [review]
Good work! A few things to improve

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/63c7708b959ed8fb9beda4722e1402c20735226c

File: app/templates/components/menu-button.html (hunk 3 above)
Comment (jimleroyer) [reply app/templates/components/menu-button.html:10]: It's a good idea, I've implemented it.

File: app/templates/partials/nav/gc_header.html (hunk 2 above)
Comment (jimleroyer) [reply app/templates/partials/nav/gc_header.html:16]: We removed the link to the GoC logo.

File: app/templates/components/nav_menu_item.html (hunk 1 above)
Comment (jimleroyer) [reply app/templates/components/nav_menu_item.html:21]: It should not jump anymore; I used your idea of a transparent border to fix it.

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/1a06031c144f5e411a2bbc1119c2fb6f570343d9

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/cc3afb0b0ada0c8286830139f83fed23ddaac0a4

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/7f969f50838838b3922f6f94d9e1e16c9136059d

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/11865c490b7cb14c0540f33a6257b20d4417bfcf

**Review by AntoineAugusti — APPROVED** [review]
Tested and looks neat

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/6692caf94bb12a9ecae6b1376b942bcdd596227f

**Pull request merged**